File: web/src/services/userApi.ts
import api from './api';
import type { User, TodueExport, ImportFormat, ImportResponse } from '../types';

export const userApi = {
  /**
   * Get current authenticated user
   */
  async getCurrentUser(): Promise<User> {
    const response = await api.get<User>('/user/me');
    return response.data;
  },

  /**
   * Get current date in user's timezone
   */
  async getCurrentDate(): Promise<string> {
    const response = await api.get<{ currentDate: string }>('/user/current-date');
    return response.data.currentDate;
  },

  /**
   * Update user's timezone
   */
  async updateTimezone(timezone: string): Promise<User> {
    const response = await api.put<User>('/user/timezone', { timezone });
    return response.data;
  },

  /**
   * Update user's accent color
   */
  async updateAccentColor(accentColor: string | null): Promise<User> {
    const response = await api.put<User>('/user/accent-color', { accentColor });
    return response.data;
  },

  /**
   * Export all user data as JSON
   */
  async exportData(): Promise<TodueExport> {
    const response = await api.get<TodueExport>('/user/export');
    return response.data;
  },

  /**
   * Import data from external source
   */
  async importData(format: ImportFormat, data: unknown): Promise<ImportResponse> {
    const response = await api.post<ImportResponse>('/user/import', { format, data });
    return response.data;
  },
};

File: mobile/lib/services/user_api.dart
import '../models/user.dart';
import 'api_service.dart';

class UserApi {
  final ApiService _apiService;

  static UserApi get instance => userApi;

  UserApi(this._apiService);

  /// Get current user information
  Future<User> getCurrentUser() async {
    final response = await _apiService.get('/user/me');
    return User.fromJson(response.data as Map<String, dynamic>);
  }

  /// Get current date in user's timezone
  Future<String> getCurrentDate() async {
    final response = await _apiService.get('/user/current-date');
    return response.data['currentDate'] as String;
  }

  /// Update user's timezone
  Future<User> updateTimezone({required String timezone}) async {
    final response = await _apiService.put(
      '/user/timezone',
      queryParameters: {
        'timezone': timezone,
      },
    );
    return User.fromJson(response.data as Map<String, dynamic>);
  }

  /// Update user's accent color
  Future<User> updateAccentColor({required String? accentColor}) async {
    final response = await _apiService.put(
      '/user/accent-color',
      data: {'accentColor': accentColor},
    );
    return User.fromJson(response.data as Map<String, dynamic>);
  }

  /// Get available timezones
  Future<List<String>> getTimezones() async {
    final response = await _apiService.get('/user/timezones');
    return (response.data as List).cast<String>();
  }
}

// Singleton instance
final userApi = UserApi(apiService);
